students/13th/hamjunho/js/main.js

```diff
@@ -0,0 +1,289 @@
+  const atag = document.createElement('a')
+  commentprofile.appendChild(atag).innerText = v.nickname
+  const spantag = document.createElement('span')
+  commentprofile.appendChild(spantag).innerText = v.contents
+  const removeBtn = document.createElement('div')
+  commentbtns.appendChild(removeBtn).classList.add('comment_remove_btn')
+  removeBtn.innerText = '삭제'
+  const heartToggleBtn = document.createElement('div')
+  commentbtns.appendChild(heartToggleBtn).classList.add('heart_toggle')
+  const whiteHeartImage = document.createElement('img')
+  whiteHeartImage.classList.add('whiteHeart')
+  whiteHeartImage.setAttribute('src','https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/heart.png')
+  heartToggleBtn.appendChild(whiteHeartImage)
+  CommentsList.appendChild(commentli)
+}
+
+const CommentsList = document.querySelector('.comments_list ul');
+dummyPost.comments.map((v) => {
+  CreateComment(v)
+})
+
+const CommentsLength = document.querySelector('.comment_length');
+CommentsLength.textContent = `댓글 ${CommentsList.children.length}개 모두 보기`
+
+const CommentFormArea = document.querySelector('.comment_form textarea');
+const CommentFormButton = document.querySelector('.comment_form button');
+function submitComment() { // 댓글 전송
+  if (CommentFormArea.value.length < 1) {
+    alert('댓글을 입력해라!')
+    return;
+  }
+  dummyPost.comments.push({
+    nickname: me.nickname,
+    contents: CommentFormArea.value,
+    liked: false
+  })
+  CommentFormArea.value = ""
+  CommentsList.innerHTML = ''
+  dummyPost.comments.map((v) => {
+    CreateComment(v)
+  })
+  CommentsLength.textContent = `댓글 ${CommentsList.children.length}개 모두 보기`
+  CommentFormButton.setAttribute('disabled', null)
+  removeComment()
+  heartToggleWith()
+}
+
+const removeComment = () => {
+  const commentRemoveBtn = document.querySelectorAll('.comment_remove_btn');
+  _map((v) => v.addEventListener('click', () => {
+    CommentsList.removeChild(v.parentNode.parentNode)
+    CommentsLength.textContent = `댓글 ${CommentsList.children.length}개 모두 보기`
+  })
+    , commentRemoveBtn)
+}
+
+removeComment()
+function heartToggleWith() {
+const heartBtn = document.querySelectorAll('.heart_toggle');
+_map((v) => v.addEventListener('click', function() {
+  const redHeartImage = document.createElement('img')
+  redHeartImage.classList.add('redHeart')
+  redHeartImage.setAttribute('src','https://www.pngfind.com/pngs/m/101-1010488_instagram-heart-png-love-heart-transparent-png.png')
+  const whiteHeartImage = document.createElement('img')
+  whiteHeartImage.classList.add('whiteHeart')
+  whiteHeartImage.setAttribute('src','https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/heart.png')
+  console.log(this.children[0].classList.value)
+  if(this.children[0].classList.value === 'whiteHeart') {
+    v.replaceChild(redHeartImage,this.children[0])
+  } else {
+    v.replaceChild(whiteHeartImage,this.children[0])
+
+  }
+}),heartBtn)
+}
+
+heartToggleWith()
```

Comment:
```suggestion
(function heartToggleWith() {
const heartBtn = document.querySelectorAll('.heart_toggle');
_map((v) => v.addEventListener('click', function() {
  const redHeartImage = document.createElement('img')
  redHeartImage.classList.add('redHeart')
  redHeartImage.setAttribute('src','https://www.pngfind.com/pngs/m/101-1010488_instagram-heart-png-love-heart-transparent-png.png')
  const whiteHeartImage = document.createElement('img')
  whiteHeartImage.classList.add('whiteHeart')
  whiteHeartImage.setAttribute('src','https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/heart.png')
  console.log(this.children[0].classList.value)
  if(this.children[0].classList.value === 'whiteHeart') {
    v.replaceChild(redHeartImage,this.children[0])
  } else {
    v.replaceChild(whiteHeartImage,this.children[0])

  }
}),heartBtn)
})();
```

- 즉시실행함수(IIFE)를 적용해볼 수 있겠네요. 검색해서 학습해보시기 바랍니다 :)

Reply:
즉시 실행 함수를 사용하면 submitComment() 안에서 heartToggleWith() 호출이 되지 않는 문제가 있는 것 같습니다.